plugins/lang/js/js_common.cc

```diff
@@ -7,27 +7,185 @@
 */
 
 #include "js_common.h"
+#include "js_utils.h"
 #include "mldb/sql/cell_value.h"
 #include "mldb/sql/expression_value.h"
 #include "mldb/types/basic_value_descriptions.h"
 #include "mldb_js.h"
+#include "mldb/ext/v8-cross-build-output/include/libplatform/libplatform.h"
+#include "mldb/base/thread_pool.h"
 #include <boost/algorithm/string.hpp>
 #include <boost/regex.hpp>
-
+#include <boost/filesystem.hpp>
+#include <thread>
 
 using namespace std;
 
+using namespace v8;
+
 
 namespace Datacratic {
 namespace MLDB {
 
 Logging::Category mldbJsCategory("javascript");
 
+/**
+ * V8 Platform abstraction layer.
+ *
+ * The embedder has to provide an implementation of this interface before
+ * initializing the rest of V8.
+ */
+struct V8MldbPlatform: public v8::Platform {
+    V8MldbPlatform(MldbServer * server)
+        : start(std::chrono::steady_clock::now())
+    {
+    }
+
+    std::chrono::time_point<std::chrono::steady_clock> start;
+    MldbServer * server;
+
+    /**
+     * Gets the number of threads that are used to execute background tasks. Is
+     * used to estimate the number of tasks a work package should be split into.
+     * A return value of 0 means that there are no background threads available.
+     * Note that a value of 0 won't prohibit V8 from posting tasks using
+     * |CallOnBackgroundThread|.
+     */
+    virtual size_t NumberOfAvailableBackgroundThreads() override
+    {
+        return ThreadPool::instance().numThreads();
+    }
+
+    /**
+     * Schedules a task to be invoked on a background thread. |expected_runtime|
+     * indicates that the task will run a long time. The Platform implementation
+     * takes ownership of |task|. There is no guarantee about order of execution
+     * of tasks wrt order of scheduling, nor is there a guarantee about the
+     * thread the task will be run on.
+     */
+    virtual void CallOnBackgroundThread(Task* task,
+                                        ExpectedRuntime expected_runtime) override
+    {
+        std::shared_ptr<Task> taskPtr(task);
+        auto lambda = [=] ()
+            {
+                taskPtr->Run();
+            };
+
+        if (expected_runtime == Platform::kShortRunningTask) {
+            ThreadPool::instance().add(lambda);
+        }
+        else {
+            std::thread(lambda).detach();
+        }
+    }
+
+    /**
+     * Schedules a task to be invoked on a foreground thread wrt a specific
+     * |isolate|. Tasks posted for the same isolate should be execute in order of
+     * scheduling. The definition of "foreground" is opaque to V8.
+     */
+    virtual void CallOnForegroundThread(Isolate* isolate, Task* task) override
+    {
+        cerr << "CallOnForegroundThread" << endl;
+        std::terminate();
```

Comment:
v8 is compiled with exceptions and RTTI off, and this is called from there, so it will have the same effect.